services/media/service/src/domains/movies/localization/movies-replication-handlers.ts

```diff
@@ -0,0 +1,85 @@
+import { Dict, isEmptyObject } from '@axinom/mosaic-service-common';
+import { movies } from 'zapatos/schema';
+import { Config } from '../../../common';
+import {
+  assertLocalizationColumn,
+  assertLocalizationData,
+  getDeleteMessageData,
+  getUpsertMessageData,
+  ReplicationOperationHandlers,
+} from '../../common';
+import { LOCALIZATION_MOVIE_TYPE } from './constants';
+import { MovieFieldDefinitions } from './get-movie-localization-entity-definitions';
+
+type LocalizableMovie = Pick<movies.JSONSelectable, 'id' | 'title'> & {
+  [k: string]: unknown;
+};
+
+const assertMovie: (
+  data: Dict<unknown> | undefined,
+) => asserts data is LocalizableMovie = (
+  data: Dict<unknown> | undefined,
+): asserts data is LocalizableMovie => {
+  assertLocalizationData(data, 'movies');
+  assertLocalizationColumn(data, 'id', 'movies');
+  assertLocalizationColumn(data, 'title', 'movies');
+};
+
+export const moviesReplicationHandlers = (
+  config: Config,
+): ReplicationOperationHandlers => {
+  const entityType = LOCALIZATION_MOVIE_TYPE;
+  const fieldDefinitions = MovieFieldDefinitions.filter((d) => !d.is_archived);
+  return {
+    insertHandler: async (newData: Dict<unknown> | undefined) => {
+      assertMovie(newData);
+
+      const fields: Dict<unknown> = {};
+      for (const { field_name } of fieldDefinitions) {
+        const value = newData[field_name];
+        if (value) {
```

Comment:
An if-check in this way is likely not right. It checks for all null-ish values and empty string vs. undefined/null makes a difference that should also be sent. As for the movie genres handler such a check is also not there.

Reply:
- This would be true for `updateHandler`, and there such check is not present,
  so whatever is set - we send out.
- When we insert the movies, we only care about sending the data to the localization
  service only for field with actual values (e.g. just title, or title +
  description + synopsis).
- For movie genres such a check is not needed, as movie genres only have the
  required title field (opposed to movies which also has synopsis/description).
  - Ofc, such check can be added for movie genres to make it future-proof in
    case it gets expanded with optional localizable fields, but I would handle
    that when/if we get there.